docs/contributing.md

```diff
@@ -117,31 +117,137 @@ fetch master and create these branches for you:
+Commit your changes using the Conventional Commits format. All commits must follow the format described in the [Conventional Commits](#conventional-commits) section below and include proper git trailers for issue tracking.
 
+**Basic commit format:**
 ```bash
-git commit -m "<component>: <change_message> #<issue number>"
+git commit -m "<type>(<scope>): <description>" --trailer "Closes: #<issue_number>"
 ```
 
-Valid component names are listed in the [__label
-list__](https://github.com/rucio/rucio/labels) and are usually specified on the
-issue of the change.
+**Example:**
+```bash
+git commit -m "feat(Core): Add new rule evaluation engine" --trailer "Closes: #1234"
+```
 
 Add additional explanations to the body of the commit, such as motivation for
 certain decisions and background information. [Here are some general rules.](https://cbea.ms/git-commit/).
 
-If you add a [__github-recognised
-keyword__](https://help.github.com/articles/closing-issues-using-keywords/) then
-the associated issue can be closed automatically once the pull request is
-merged, e.g.:
+Using multiple commits is allowed as long as they achieve an independent,
+well-defined, change and are well-described. Otherwise multiple commits should
+be squashed.
+
+### **Conventional Commits**
+
+Rucio enforces the [Conventional Commits](https://www.conventionalcommits.org/) specification to ensure consistent and meaningful commit messages across the project. This is enforced through [commitlint](https://commitlint.js.org/) during CI checks and can be enabled locally via pre-commit hooks.
+
+**Commit Message Format:**
+```
+<type>(<scope>): <description>
+
+[optional body]
+
+[optional footer(s)]
+```
 
+**Rules:**
+
+1. **Type**: Must be one of the following allowed types:
+   - `feat`: New feature
+   - `fix`: Bug fix
+   - `perf`: Code change that improves performance
+   - `docs`: Documentation only changes
+   - `style`: Changes that do not affect the meaning of the code
+   - `refactor`: Code change that neither fixes a bug nor adds a feature
+   - `test`: Adding missing tests or correcting existing tests
+   - `build`: Changes that affect the build system or external dependencies
+   - `ci`: Changes to CI configuration files and scripts
+   - `chore`: Other changes that don't modify src or test files
+   - `revert`: Reverts a previous commit
+
+2. **Scope**: Must be one of the predefined Rucio components (PascalCase). The available scopes are:
+   - `Auth`: Authentication & Authorisation
+   - `Clients`: Client libraries and tools
+   - `Consistency`: Consistency checks
+   - `CI`: Continous Integration
+   - `Core`: Core & Internals
+   - `Database`: Database-related changes
+   - `DatasetDeletion`: Dataset deletion functionality
+   - `Deletion`: File deletion functionality
+   - `DIRAC`: DIRAC integration
+   - `Docker`: Docker related functionality
+   - `Documentation`: Documentation updates
+   - `Kubernetes`:  Kubernetes deployment related functionality
```

Comment:
Docker and Kubernetes are part of a single component https://rucio.github.io/documentation/component_leads/ - I would also be in favour of maybe just calling it `Containerization` so that we don't have to use tool-specific names